pkg/crypto/keys/secp256k1.go

```diff
@@ -0,0 +1,36 @@
+package keys
+
+import (
+	"crypto/elliptic"
+	"math/big"
+	"sync"
+)
+
+// Secp256k1Curve represents the name of Secp256k1 elliptic curve, which is set to be "Secp256k1".
+const Secp256k1Curve string = "Secp256k1"
+
+type secp256k1Curve struct {
+	*elliptic.CurveParams
+}
+
+var (
+	initonce  sync.Once
+	secp256k1 secp256k1Curve
+)
+
+func initSecp256k1() {
+	secp256k1.CurveParams = &elliptic.CurveParams{Name: Secp256k1Curve}
+	secp256k1.P, _ = new(big.Int).SetString("fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f", 16)
+	secp256k1.N, _ = new(big.Int).SetString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141", 16)
+	secp256k1.B, _ = new(big.Int).SetString("0000000000000000000000000000000000000000000000000000000000000007", 16)
+	secp256k1.Gx, _ = new(big.Int).SetString("79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798", 16)
+	secp256k1.Gy, _ = new(big.Int).SetString("483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8", 16)
+	secp256k1.BitSize = 256
+}
+
+// Secp256k1 returns a Curve which implements secp256k1 elliptic curve.
+// The CurveParams.Name of this Curve is "Secp256k1".
+func Secp256k1() elliptic.Curve {
+	initonce.Do(initSecp256k1)
```

Comment:
Why not do this in `init()`?

Reply:
And if we use `btcec` package, we do not need this at all.

Reply:
That was a draft, now removed this code. We should also decide which package to use for Koblitz.